microk8s/build-and-release-on-new-upstream-release.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/python3
+
+import requests
+import configbag
+import canonicalwebteam.snapstoreapi.public_api as public_api
+from launchpadlib.launchpad import Launchpad
+from lazr.restfulclient.errors import HTTPError
+
+
+tracks = ["latest", "1.10", "1.11", "1.12", "1.13", "1.14", "1.15"]
+
+
+def upstream_release(release):
+    """Return the latest stable k8s in the release series"""
+    if release == "latest":
+        release_url = "https://dl.k8s.io/release/stable.txt"
+    else:
+        release_url = "https://dl.k8s.io/release/stable-{}.txt".format(release)
+
+    r = requests.get(release_url)
+    if r.status_code == 200:
+        return r.content.decode().strip()
+    else:
+        None
+
+
+def snapped_release(track):
+    """ Return the version of the microk8s snap in the edge channel and the track provided"""
+    snap_details = public_api.get_snap_details('microk8s', 'edge')
+    tracks = snap_details['channel_maps_list']
+    versions = [c['version'] for t in tracks if t['track'] == track
+                for c in t['map'] if c['channel'] == 'edge']
+    version = versions[0] if versions else None
+    return version
+
+
+def trigger_lp_builders(track):
+    """Trigger the LP builder of the track provided. This method will
+    login using the cached credentials or prompt you for authorization."""
+    if track == "latest":
+        snap_name = "microk8s"
+    else:
+        snap_name = "microk8s-{}".format(track)
+
+    # log in
+    launchpad = Launchpad.login_with('Launchpad Snap Build Trigger',
+                                     'production', configbag.cachedir,
+                                     credentials_file=configbag.creds,
+                                     version='devel')
+
+    # get launchpad team data and ppa
+    snappydev = launchpad.people[configbag.people_name]
+
+    try:
+        # get snap
+        microk8s = launchpad.snaps.getByName(name=snap_name,
+                                               owner=snappydev)
+    except HTTPError as e:
```

Comment:
Why is this the only place where an `HTTPError` is caught?  Wouldn't other API calls potentially raise errors as well?  Do we even need to bother catching this one?

Reply:
This is an exception which is somewhat expected. We may get this exception in case we do not have builders for this specific release. For example for 1.12 we may not create the builders immediately, so instead of raising the exception and failing the build we treat this error as an expected one. This is not 100% safe as we may be shadowing some true exceptions but I would expect any sporadic HTTP errors to be few and any failed builders builds to be triggered in the next cycle.